### PR TITLE
Improved samplesheet generation for NovaSeqX.

### PR DESCRIPTION
**Pull request by mgcam**
Commented out code deleted.
Attributes and methods are ordered more logically. Some private accesors are converted to public.
Documentation improved.
Added index_read_length and read_length attributes so that these values can be set by the caller.

### CODE REVIEW

File: lib/npg/samplesheet/novaseq_xseries.pm [hunk 1]
```diff
@@ -158,9 +158,60 @@ sub _build_batch_id {
...
+
+=cut
+
+has 'index_read_length' => (
+  'isa'        => 'ArrayRef',
+  'is'         => 'ro',
+  'lazy_build' => 1,
+  'required'   => 0,
+  'documentation' => 'An array containing the length of the first and the ' .
+                     'second (if applicable) indexing read..',
```
Comment (jmtcsngr) [inline lib/npg/samplesheet/novaseq_xseries.pm:178]:
```suggestion
                     'second (if applicable) indexing read.',
```

File: lib/npg/samplesheet/novaseq_xseries.pm [hunk 2]
```diff
@@ -158,9 +158,60 @@ sub _build_batch_id {
...
+
+An array containing the length of the first and the second (if applicable)
+indexing read.
+
+If not set, is computed as the longest first and second barcode as reported
+by the LIMS system.
+
+=cut
+
+has 'index_read_length' => (
```
Comment (jmtcsngr) [inline lib/npg/samplesheet/novaseq_xseries.pm:172]: Is this meant to be used as an override? Should we care if only one is provided at executing time but there are two lengths in LIMS?

**Review by jmtcsngr — COMMENTED** [review]
Only suggestions and a couple of questions.

File: lib/npg/samplesheet/novaseq_xseries.pm (hunk 2 above)
Comment (mgcam) [reply lib/npg/samplesheet/novaseq_xseries.pm:172]: This is when we want to do something semi-manually. Prior to the start of the run we only have LIMS information. If we want to run DRAGEN off-board, then we will find out what happenned from RunInfo.xml As we know only two well, all sorts of discrepancies are possible.

**Pull request merged**